spotbugs/src/main/java/edu/umd/cs/findbugs/detect/AbstractAssertDetector.java

```diff
@@ -0,0 +1,51 @@
+package edu.umd.cs.findbugs.detect;
+
+import org.apache.bcel.Const;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+
+/**
+ * Abstract base class for finding assertions
+ */
+public abstract class AbstractAssertDetector extends OpcodeStackDetector {
+
+    private final BugReporter bugReporter;
+
+    protected boolean inAssert = false;
+
+    public AbstractAssertDetector(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
+    /**
+     * Implement this method in a concrete detector
+     */
+    abstract protected void detect(int seen);
+
+    /**
+     * Searches for assertion opening, and closing points.
+     * When in assert, will call the detect method.
+     */
+    @Override
+    public void sawOpcode(int seen) {
+        if (inAssert) {
+            detect(seen);
+        }
+        if (seen == Const.GETSTATIC && "$assertionsDisabled".equals(getNameConstantOperand())) {
+            inAssert = true;
+        }
+        if (seen == Const.NEW && getClassConstantOperand().equals("java/lang/AssertionError")) {
+            resetState();
```

Comment:
I think this one can be resolved, since it was solved in [commit fe8cb9af72ba8078b920d74a2a776a2952d0db16].(https://github.com/spotbugs/spotbugs/pull/2130/commits/fe8cb9af72ba8078b920d74a2a776a2952d0db16)